packages/webui/src/client/ui/SegmentTimeline/SegmentTimelineContainer.tsx

```diff
@@ -136,11 +142,39 @@ export function SegmentTimelineContainer(props: Readonly<IProps>): JSX.Element {
 	// past infinites subscription
 	useSubscription(CorelibPubSub.piecesInfiniteStartingBefore, props.rundownId, sortedSegmentIds, sortedRundownIds)
 
-	return <SegmentTimelineContainerContent {...props} />
+	return <SegmentTimelineContainerWithSelection {...props} />
+}
+
+function SegmentTimelineContainerWithSelection(props: Readonly<IProps>): JSX.Element {
+	const { clearAndSetSelection, isSelected } = useSelection()
+
+	const handleSegmentSelect = (segmentId: SegmentId) => {
+		console.log('handleSegmentSelect', segmentId)
+		clearAndSetSelection({
+			type: 'segment',
+			id: String(segmentId),
+			elementId: segmentId,
+		})
+	}
+
+	const handleIsSelected = (segmentId: SegmentId) => {
+		return isSelected(String(segmentId))
+	}
+
+	return (
+		<SegmentTimelineContainerContent
+			{...props}
+			handleIsSelected={handleIsSelected}
```

Comment:
Why is this injected into the component as a function? `SegmentTimelineContainerWithSelection` seems to be completely in control of the selection state of the Segment. This component will re-render anyway whenever the context changes (since `useSelection` will need to return), and it is the thing responsible for creating the `id` of the Selection - wouldn't it make sense to just inject a plain boolean here?

This current setup creates a tangled web of relationships between who's providing what to whom. A plain callback and a plain boolean value would suffice, I think?

Reply:
Agree and if it wasn't because of that terrible withResolvedSegment it would probably not had been added in first place.
Will clean up.